openesdh-repo/src/main/java/dk/openesdh/repo/services/audit/AuditSearchServiceImpl.java

```diff
@@ -79,67 +64,51 @@ public JSONArray getAuditLogByCaseNodeRef(NodeRef nodeRef, int timespan) {
-        }
-        return false;
+        return (currentUserGroups.stream().anyMatch((userGroup) -> (rwoGroups.contains(userGroup))));
     }
 
     private Set<String> getCurrentUserGroups() {
-        return AuthenticationUtil.runAs(new AuthenticationUtil.RunAsWork<Set<String>>() {
-            @Override
-            public Set<String> doWork() throws Exception {
-                String currentUserName = AuthenticationUtil.getFullyAuthenticatedUser();
-                Set<String> userGroups = authorityService.getAuthoritiesForUser(currentUserName);
-                return userGroups;
-            }
+        return AuthenticationUtil.runAs(() -> {
+            String currentUserName = AuthenticationUtil.getFullyAuthenticatedUser();
+            Set<String> userGroups = authorityService.getAuthoritiesForUser(currentUserName);
+            return userGroups;
         }, AuthenticationUtil.getAdminUserName());
     }
 
-    private List<String> getCaseReadWriteOwnGroups(NodeRef nodeRef){
+    private List<String> getCaseReadWriteOwnGroups(NodeRef nodeRef) {
         List<String> rwoPermissions = Arrays.asList(new String[]{
-                OpenESDHModel.PERMISSION_NAME_CASE_SIMPLE_READER,
-                OpenESDHModel.PERMISSION_NAME_CASE_SIMPLE_WRITER,
-                OpenESDHModel.PERMISSION_NAME_CASE_OWNERS
+            OpenESDHModel.PERMISSION_NAME_CASE_SIMPLE_READER,
+            OpenESDHModel.PERMISSION_NAME_CASE_SIMPLE_WRITER,
+            OpenESDHModel.PERMISSION_NAME_CASE_OWNERS
         });
-        ArrayList<String> rwoGroups = new ArrayList<String>();
+        ArrayList<String> rwoGroups = new ArrayList<>();
         Set<AccessPermission> casePermissions = getAllCasePermissions(nodeRef);
-        for(AccessPermission accessPermission : casePermissions){
-            if (accessPermission.getAuthorityType() == AuthorityType.GROUP
-                    && rwoPermissions.contains(accessPermission.getPermission())) {
-                rwoGroups.add(accessPermission.getAuthority());
-            }
-        }
+        casePermissions.stream()
+                .filter((accessPermission)
+                        -> (accessPermission.getAuthorityType() == AuthorityType.GROUP
+                        && rwoPermissions.contains(accessPermission.getPermission())))
+                .forEach((accessPermission) -> {
+                    rwoGroups.add(accessPermission.getAuthority());
+                });
         return rwoGroups;
     }
 
     private Set<AccessPermission> getAllCasePermissions(final NodeRef nodeRef) {
-        return AuthenticationUtil.runAs(
-                new AuthenticationUtil.RunAsWork<Set<AccessPermission>>() {
-                    @Override
-                    public Set<AccessPermission> doWork() throws Exception {
-                        return permissionService.getAllSetPermissions(nodeRef);
-                    }
-                }, AuthenticationUtil.getAdminUserName());
+        return AuthenticationUtil.runAs(()
```

Comment:
This "runAs admin" could be extracted into separate method to avoid code duplication.